src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginUpdater.java

```diff
@@ -0,0 +1,289 @@
+package com.github.manolo8.darkbot.extensions.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.gui.plugins.UpdateProgressBar;
+import com.github.manolo8.darkbot.gui.utils.Popups;
+import com.github.manolo8.darkbot.utils.Time;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+//todoo i18n
+public class PluginUpdater {
+
+    private Date lastChecked = new Date();
+
+    public final List<Plugin> UPDATES = new ArrayList<>();
+
+    private final PluginHandler pluginHandler;
+
+    private UpdateProgressBar mainProgressBar;
+    private final List<PluginProgress> pluginProgresses = new ArrayList<>();
+
+    private static class PluginProgress {
+        private final Plugin plugin;
+        private final UpdateProgressBar progressBar;
+        private final JLabel progressLabel;
```

Comment:
Feels like ur holding a bit too much of GUI on logic-code

Consider having just PluginCard (s) and having a method there that is setUpdateProgress(x), that automatically update progress bar, sets it visible or invisible, as well as the label etc